### PR TITLE
Dynamodb has been set up

### PR DESCRIPTION
**Pull request by humulla**
## Link to JIRA ticket

https://hackney.atlassian.net/jira/software/projects/MTTL/boards/47?selectedIssue=MTTL-492 

## Describe this PR

The following tasks have been completed:

- [x] Set up terraform for all three environments
- [x] Set up dynamodb for all three environments
- [x] Set the partition (hash) key as targetId
- [x] Set the sort (range) key as id

### CODE REVIEW

**Review by mikemoody-amido — COMMENTED** [review]
The terraform suggestions on the dev file should also apply to the other 2 environments.

In addition...
The `appsettings.json` file needs to set the AWS region to be: **eu-west-2**
...otherwise when deployed the code will not be able to connect to DynamoDb.

File: terraform/development/dynamodb.tf [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    read_capacity         = 10
+    write_capacity        = 10
+    hash_key              = "targetId"
+    range_key             = "id"
+
+
+    attribute {
+        name              = "id"
+        type              = "S"
+    }
```
[the remark this answers is not in the document]
Comment (humulla) [reply terraform/development/dynamodb.tf:13]: Thanks for spotting this, I've now added it in

**Comment by humulla** [comment]
@mikemoody-amido, Thank you for your feedback I've made the changes you've recommended let me know if it's okay

File: terraform/development/dynamodb.tf [hunk 2]
```diff
@@ -1,5 +1,5 @@
 resource "aws_dynamodb_table" "contactdetailsapi_dynamodb_table" {
-    name                  = "contact-details"
+    name                  = "Contact Details"
```
Comment (mikemoody-amido) [inline terraform/development/dynamodb.tf:2]: This needs to be **ContactDetails** (no spaces)

Comment (humulla) [reply terraform/development/dynamodb.tf]: I've removed the space

**Review by mikemoody-amido — APPROVED** [review]

**Pull request merged**